src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/TaskController.kt

```diff
@@ -97,38 +97,61 @@ class TaskController {
     }
 
     /**
-     * Lists tasks
+     * Lists tasks by milestones
      *
-     * @param project project
+     * @param milestones milestones
      * @return list of tasks
      */
     suspend fun list(milestones: List<MilestoneEntity>): List<TaskEntity> {
         return taskEntityRepository.find("milestone in :milestones", Parameters().and("milestones", milestones))
             .list<TaskEntity>().awaitSuspending()
     }
 
+    /**
+     * Lists tasks by dependent user
+     *
+     * @param dependentUser dependent user
+     * @return list of tasks
+     */
+    suspend fun list(dependentUser: UserEntity): List<TaskEntity> {
+        return taskEntityRepository.find("dependentUser", dependentUser).list<TaskEntity>().awaitSuspending()
+    }
+
     /**
      * Creates a new task and triggers the needed notifications, assigns users to the project if not yet done
      *
      * @param milestone milestone
      * @param task task
      * @param userId user id
      * @return created task
-     * @throws UserNotFoundException if assignee is not found
+     * @throws UserNotFoundException if assignee or dependent user is not found
      */
     suspend fun create(
         milestone: MilestoneEntity,
         jobPosition: JobPositionEntity?,
         task: Task,
         userId: UUID
     ): TaskEntity {
+        val dependentUser = if (task.dependentUserId != null) {
+            val user = userController.findUser(task.dependentUserId) ?: throw UserNotFoundException(task.dependentUserId)
+            if (!projectController.hasAccessToProject(milestone.project, user.keycloakId)) {
+                logger.info("Assigning user ${user.id} to project ${milestone.project.id} because of the task assignment")
+                userController.assignUserToProjects(
+                    user = user,
+                    newProjects = listOf(milestone.project)
+                )
+            }
+            user
+        } else null
+
```

Comment:
```suggestion
        val dependentUser = task.dependentUserId?.let {
            val user = userController.findUser(task.dependentUserId) ?: throw UserNotFoundException(task.dependentUserId)
            if (!projectController.hasAccessToProject(milestone.project, user.keycloakId)) {
                logger.info("Assigning user ${user.id} to project ${milestone.project.id} because of the task assignment")
                userController.assignUserToProjects(
                    user = user,
                    newProjects = listOf(milestone.project)
                )
            }
            user
        }

```

Reply:
Fixed